parse.go

```diff
@@ -293,29 +285,72 @@ func getUpdatedLock() (*Protolock, error) {
 		if !strings.HasSuffix(info.Name(), protoSuffix) {
 			return nil
 		}
+
 		// skip to next if is a directory
 		if info.IsDir() {
 			return nil
 		}
 
+		// if ignore, skip
+		if ignores != "" {
+			for _, ignore := range strings.Split(ignores, ",") {
+				rootIgnore := root + "/" + ignore
+				stat, err := os.Stat(rootIgnore)
+				if err == nil {
+					if stat.IsDir() && !strings.HasSuffix(rootIgnore, "/") {
+						rootIgnore += "/"
+					}
+
+					if strings.HasPrefix(path, rootIgnore) {
```

Comment:
https://golang.org/pkg/path/filepath/#HasPrefix

Reply:
HasPrefix exists for historical compatibility and should not be used.

Deprecated: HasPrefix does not respect path boundaries and does not ignore case when required.